resource/rabbitmq/rabbitmq.go

```diff
@@ -6,96 +6,120 @@ import (
+	resourceName        = "rabbitmq"
 )
 
 type RabbitMQ struct {
 	mtx             sync.Mutex
+	datasource      string
 	conn            *amqp.Connection
 	consumedMessage map[string][][]byte
 	waitDuration    time.Duration
 }
 
+// New creates and validates the resource params for the connection initialized
+// in Open()
 func New(cfg *config.Resource) (*RabbitMQ, error) {
 	params := cfg.Params
 	datasource, ok := params["datasource"]
 	if !ok {
-		return nil, errors.New("queue/rabbitmq: datasource is required")
+		return nil, errors.Errorf("%s: datasource is required", resourceName)
 	}
 
 	waitDuration, err := time.ParseDuration(params["wait_duration"])
 	if err != nil {
 		waitDuration = DefaultWaitDuration
 	}
 
-	conn, err := amqp.Dial(datasource)
-	if err != nil {
-		return nil, errors.New("queue/rabbitmq: failed to connect > " + err.Error())
-	}
-
 	return &RabbitMQ{
-		conn:            conn,
+		datasource:      datasource,
 		waitDuration:    waitDuration,
 		consumedMessage: make(map[string][][]byte),
 	}, nil
 }
 
-func (c *RabbitMQ) target(target string) (string, string) {
-	result := strings.Split(target, ":")
-	if len(result) < 2 {
-		return result[0], ""
+// Open attempts a dial connection to the AMQP server
+func (c *RabbitMQ) Open() error {
+	var err error
+	c.conn, err = amqp.Dial(c.datasource)
+	if err != nil {
+		return errors.Wrapf(err, "%s: failed to connect on initial dial", resourceName)
 	}
-	return result[0], result[1]
+
+	return nil
 }
 
+// Ready attempts to publish a message to a test queue, if no error is returned the
+// server is deemed ready
 func (c *RabbitMQ) Ready() error {
 	if err := c.Publish("test:test", []byte("")); err != nil {
-		return errors.Wrapf(err, "queue: rabbitmq is not ready")
+		return errors.Wrapf(err, "%s: ready check failed", resourceName)
 	}
 	return nil
 }
 
+// Reset empties the consumedMessage map and cleans up the queues
 func (c *RabbitMQ) Reset() error {
+	for target := range c.consumedMessage {
+		exchange, key := c.target(target)
+		ch, err := c.conn.Channel()
+		if err != nil {
+			return errors.Wrapf(err, "%s: reset connection attempt failed", resourceName)
+		}
+		if _, err := ch.QueueDelete(c.queueName(exchange, key), false, false, false); err != nil {
+			return errors.Wrapf(err, "%s: reset attempt failed `%s`", resourceName, c.queueName(exchange, key))
+		}
+	}
 	c.consumedMessage = make(map[string][][]byte)
 	return nil
 }
 
+// Close cleans up the underlying connection
+func (c *RabbitMQ) Close() error {
+	if err := c.conn.Close(); err != nil {
+		return errors.Wrapf(err, "%s: failed to close connection", resourceName)
+	}
+	return nil
+}
+
+// Listen creates a consumer on the passed target, creates the queue, and puts the messages
+// into consumedMessage map for later access
 func (c *RabbitMQ) Listen(target string) error {
 	ch, err := c.conn.Channel()
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "%s: listen attempt failed", resourceName)
```

Comment:
suggestion: this can be a private function that returned prefixed resource name, for the same of consistency.